src/client/v2/indexer/searchForApplications.ts

```diff
@@ -1,36 +1,123 @@
 import JSONRequest from '../jsonrequest';
 
+/**
+ * Returns information about indexed applications.
+ *
+ * #### Example
+ * ```typescript
+ * const apps = await indexerClient.searchForApplications().do();
+ * ```
+ *
+ * [Response data schema details](https://developer.algorand.org/docs/rest-apis/indexer/#get-v2applications)
+ * @category GET
+ */
 export default class SearchForApplications extends JSONRequest {
+  /**
+   * @returns `/v2/applications`
+   */
   // eslint-disable-next-line class-methods-use-this
   path() {
     return '/v2/applications';
   }
 
-  // application ID for filter, as int
+  /**
+   * Application ID for filter, as int
+   *
+   * #### Example
+   * ```typescript
+   * const appId = 60553466;
+   * const apps = await indexerClient
+   *        .searchForApplications()
+   *        .index(appId)
+   *        .do();
+   * ```
+   * @remarks Alternatively, use `indexerClient.lookupApplications(appId).do()`
+   * @param index
+   */
   index(index: number) {
     this.query['application-id'] = index;
     return this;
   }
 
-  // creator for filter, as string
+  /**
+   * Creator for filter, as string
+   *
+   * #### Example
+   * ```typescript
+   * const creator = "XBYLS2E6YI6XXL5BWCAMOA4GTWHXWENZMX5UHXMRNWWUQ7BXCY5WC5TEPA";
+   * const apps = await indexerClient
+   *        .searchForApplications()
+   *        .creator(creator)
+   *        .do();
+   * ```
+   * @param creator
+   */
   creator(creator: string) {
     this.query.creator = creator;
     return this;
   }
 
-  // token for pagination
+  /**
+   * Specify the next page of results.
+   *
+   * #### Example
+   * ```typescript
+   * const maxResults = 20;
+   * const nextToken = "APA6C7C3NCANRPIBUWQOF7WSKLJMK6RPQUVFLLDV4U5WCQE4DEF26D4E3E";
```

Comment:
This is not at all a requirement, but my suggestion for `nextToken` docs would be to show two requests, something like this, so it's more clear where the value for `nextToken` should come from:

```javascript
const maxResults = 20;
const appsPage1 = await indexerClient
        .searchForApplications()
        .limit(maxResults)
        .do();

const appsPage2 = await indexerClient
        .searchForApplications()
        .limit(maxResults)
        .nextToken(appsPage1["next-token"])
        .do();
```